### ISSUE TITLE
helpers should be separate module

### ISSUE THREAD

**Issue by skylize** [vitaly-t/pg-promise#301]
Helpers don't need access to the database pool and do not offer any functionality back into the core. They should be pulled out into a separate module that can be required separately only when needed and don't have to carry around a full pg-promise instance to access them.

Example useage: 
*app.js*
```javascript
//...
const db = require('pg-promise')(opts)(config)
require('./model').insert(db, data, table)
//...
```
*model.js*
```javascript
module.exports = {
  insert (data, db, table){
    const query = 
      require('pg-promise-helpers')
        .insert(data, Object.keys(data), table)
    return db.query(query)
  }
}
```

**Comment by vitaly-t** [comment]
`helpers` were built entirely on the query formatting engine that's inseparable from the library. They also do benefit from the global options passed into the library itself, such as `capSQL`.

**Comment by skylize** [comment]
Ok, that makes sense. How about adding a reference to helpers to the connection object?

**Comment by vitaly-t** [comment]
> a reference to helpers to the connection object

I don't really understand what that means.

**Comment by skylize** [comment]
What I mean is that since db connection pool is already carrying around the overhead of the entire pgp module, there is no significant change if you just assign helpers to the connection pool object. But providing that reference makes it so using helpers is easier and potentially reduces overhead significantly in some common design patterns.

Returning to an altered version of the previous example with lots more comments, here is what I mean:

Example useage: 
*app.js*
```javascript
//...
// instantiate connection pool
const db = require('pg-promise')(opts)(config)
// insert some data through a data model from another file
require('./model').insert(db, data, table)
//...
```
*model.js* (currently working, but too much overhead)
```javascript
// Model would normally have some business logic and/or validation,
//   but for simplicity of example we just do the insert.
module.exports = {
   // Even though we already have the connection pool at hand,
    //    we must also pass around pgp object to access helpers 
    //    or, as done here, include & instantitate an entirely new copy of pgp.
  insert (data, db, table){
    // build query using helpers
    const query =
      require('pg-promise')().helpers
      .insert(data, Object.keys(data), table)
    // insert and return result
    return db.query(query)
  }
}
```
*model.js* (better solution, but currently won't work)
```javascript
// Model would normally have some business logic and/or validation,
//   but for simplicity of example we just do the insert.
module.exports = {
      // This would be much better, but fails because the connection 
    //    pool has no reference to helpers
  insert (data, db, table){
    // build query using helpers
    const query = 
      // Error: cannot read property 'insert' of undefined
      db.helpers.insert(data, Object.keys(data), table) 
    // insert and return result
    return db.query(query)
  }
}
```
Fix needs to be done internally as shown by this attempted patch
*app.js* (failed patch)
```javascript
//...
// instantiate connection pool
const pgp =require('pg-promise')(opts) 
const db = pgp(config)

// Attempt to attach a helpers reference to our connection pool
//    before passing it to the model.
//    TypeError: Can't add property helpers, object is not extensible
db.helpers = pgp.helpers

// insert some data through a data model from another file
require('./model').insert(db, data, table)
//...
```

**Comment by skylize** [comment]
This issue is appropriately closed. So I am going to reopen a new issue with the new request and copy this example to it.

**Comment by vitaly-t** [comment]
> const query =
      require('pg-promise')().helpers

You should never do such coding. Variable `pgp` must be created only once and then exported around.

>db.helpers

This is not a good idea, as it pollutes an already extensible protocol - event `extend`.
